docs/improving-gci-experience.md

```diff
@@ -0,0 +1,350 @@
+resources we have is the queue in the GCI app (example screenshot below).
+If you've been working on your task and asking for help on chat, you need to
+submit the task and it will be queued in the app. It's not a perfect solution,
+but keeping with the queue assures we are more objective than by simply
+attending only to students active on Zulip chat.
+
+![Example dashboard view](gci-app-screenshot.png)
+
+Other important points about the response times:
+
+* Mentors are working shifts across multiple timezones, to assure the best
+coverage - if a mentor is online, it doesn't mean they're working on the
+GCI tasks, since we also have our own tasks for Zulip when not on shift.
+
+* The [GCI roles and responsibilities](https://developers.google.com/open-source/gci/help/responsibilities)
+ have some general rules about response times:
+  > Respond to questions within 24 hours (occasionally under 36 hours is ok)
+
+* If it's been more than four hours since you've gotten attention from a mentor,
+it's reasonable to post something on your task topic that says you've been
+waiting for a while. Include useful things like your PR link, your task link,
+and a short status of where things left off with the previous mentor.
+
+* If you're not getting a quick response from a mentor, take the time to
+double-check your work or find other constructive things to do. Make sure you
+don't have idle time - note down your learning progress, seek out new challenges,
+work on a Python tutorial etc.
+
+### How a mentor shift looks like?
+
+In order of priority, mentor tasks include:
+
+* Checking the queue on the GCI app and reviewing all submitted tasks.
+
+* Answering questions on Zulip - getting into the context, looking for resources.
+
+* Checking `zulip-gci` and `zulip` repositories for relevant pull requests.
+
+## How can I ask for help?
+
+* Before you ask a question, make sure you understand the problem and try to
+solve it on your own first - we encourage a [you must try, and then you must
+ask](https://blogs.akamai.com/2013/10/you-must-try-and-then-you-must-ask.html)
+approach. Have a hypothesis on what might be the problem.
+
+* Look for answers on your own using:
+
+  * [Zulip documentation](http://zulip.readthedocs.io/en/latest/).
+
+  * Documentation for the libraries and other tools you're using.
+
+  * Simple Google searches, e.g. `python import a library`.
+
+  * [StackOverflow](http://stackoverflow.com/) and
+  [Quora](https://www.quora.com/) threads on relevant topics.
+
+* Make sure your message has all the relevant information:
+
+  * Give context - link the task you're working on, provide information on
+  expected outcome and what you're getting instead.
+
+  * Describe what you have already tried to achieve your goal.
+
+  * Make sure to add any relevant links, screenshots, code snippets.
+
+  * Format the message appropriately - use \`\`\` for long snippets and \` for
```

Comment:
Ah, right, that would just make a mess. Also, I'm not sure if all of them know it can have syntax highlighting - maybe I could point them to the Zulip shortcut prompt here? (or am I being too lazy)